File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n=len(nums)
        nMap={}
        for i in range(n):
            complement=target-nums[i]
            if complement in nMap:
                return [nMap[complement],i] #return as list the index of complement and i
            nMap[nums[i]]=i #adding the complement to the map
            
        return [] #no solution

File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
       HashMap<Integer,Integer> sum=new HashMap<>();
       for(int i=0;i<nums.length;i++)
       {
        if(sum.containsKey(target-nums[i]))
            {
                int res[]={sum.get(target-nums[i]),i};
                return res;
            }
        else
            sum.put(nums[i],i);
       }
       return new int[]{-1,-1};
}
}
